shared-bindings/tca9555/__init__.h

```diff
@@ -0,0 +1,119 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2023 Christopher Parrott for Pimoroni Ltd
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#ifndef MICROPY_INCLUDED_SHARED_BINDINGS_TCA9555___INIT___H
+#define MICROPY_INCLUDED_SHARED_BINDINGS_TCA9555___INIT___H
+
+#include "common-hal/microcontroller/Pin.h"
+
+// Type object used in Python. Should be shared between ports.
+extern const mp_obj_dict_t tca_pin_globals;
+extern const mp_obj_type_t tca_pin_type;
+
+const mcu_pin_obj_t *validate_obj_is_free_pin_including_tca(mp_obj_t obj, qstr arg_name);
+const mcu_pin_obj_t *validate_obj_is_free_pin_including_tca_or_none(mp_obj_t obj, qstr arg_name);
+const mcu_pin_obj_t *validate_obj_is_pin_including_tca(mp_obj_t obj, qstr arg_name);
+
+extern bool common_hal_tca_gpio_get_input(uint tca_gpio);
+extern bool common_hal_tca_gpio_get_output(uint tca_gpio);
+extern bool common_hal_tca_gpio_get_config(uint tca_gpio);
+extern bool common_hal_tca_gpio_get_polarity(uint tca_gpio);
+
+extern void common_hal_tca_gpio_set_output(uint tca_gpio, bool value);
+extern void common_hal_tca_gpio_set_config(uint tca_gpio, bool output);
+extern void common_hal_tca_gpio_set_polarity(uint tca_gpio, bool polarity);
```

Comment:
These should be moved to the DigitalInOut-like object because they are for a single IO.